src/app/features/charts/components/biodomains-chart/biodomains-chart.component.ts

```diff
@@ -0,0 +1,203 @@
+      svg
+        .append('g')
+        .attr('id', 'negative-bars')
+        .selectAll('rect')
+        .data(this.data)
+        .enter().append('rect')
+        .attr('x', d => labelWidth + xScale(d.pct_linking_terms))
+        .attr('y', d => yScale(d.biodomain) || 0)
+        .attr('width', d => chartWidth - labelWidth - xScale(d.pct_linking_terms))
+        .attr('height', yScale.bandwidth())
+        .attr('fill', 'white')
+        .on('click', (event) => {
+          const index = d3.select('#negative-bars').selectAll('rect').nodes().indexOf(event.currentTarget);
+          this.setStyles(index);
+        })
+        .on('mouseover', (event) => {
+          const index = d3.select('#negative-bars').selectAll('rect').nodes().indexOf(event.currentTarget);
+          d3.select('#bars').select(`rect:nth-child(${ index + 1 })`).style('fill-opacity', '100%');
+          d3.select('#bar-label').select(`text:nth-child(${ index + 1 })`).style('font-weight', 'bold');
+        })
+        .on('mouseout', (event) => {
+          const index = d3.select('#negative-bars').selectAll('rect').nodes().indexOf(event.currentTarget);
+          // if the target is the selected index
+          if (this.selectedIndex !== index) {
+            d3.select('#bars').select(`rect:nth-child(${ index + 1 })`).style('fill-opacity', '50%');
+            d3.select('#bar-label').select(`text:nth-child(${ index + 1 })`).style('font-weight', 'normal');
+          }
+        });
+
+      // bar
+      svg
+        .append('g')
+        .attr('id', 'bars')
+        .selectAll('rect')
+        .data(this.data)
+        .enter().append('rect')
+        .attr('x', labelWidth)
+        .attr('y', d => yScale(d.biodomain) || 0)
+        .attr('width', d => xScale(d.pct_linking_terms))
+        .attr('height', yScale.bandwidth())
+        .attr('fill', barColor)
+        .style('fill-opacity', d => this.selectedBioDomain === d.biodomain ? '100%' : '50%')
+        .on('click', (event) => {
+          const index = d3.select('#bars').selectAll('rect').nodes().indexOf(event.currentTarget);
+          this.setStyles(index);
+        })
+        .on('mouseover', (event) => {
+          const index = d3.select('#bars').selectAll('rect').nodes().indexOf(event.currentTarget);
+          d3.select('#bars').select(`rect:nth-child(${ index + 1 })`).style('fill-opacity', '100%');
+          d3.select('#bar-label').select(`text:nth-child(${ index + 1 })`).style('font-weight', 'bold');
+        })
+        .on('mouseout', (event) => {
+          const index = d3.select('#bars').selectAll('rect').nodes().indexOf(event.currentTarget);
+          // if the target is the selected index
+          if (this.selectedIndex !== index) {
+            d3.select('#bars').select(`rect:nth-child(${ index + 1 })`).style('fill-opacity', '50%');
+            d3.select('#bar-label').select(`text:nth-child(${ index + 1 })`).style('font-weight', 'normal');
+          }
+        });
```

Comment:
Looks like it's mostly the same as negative space handlers, I think it might reduce unnecessary duplicate code make sense to just have functions that create each handler 

```ts
function getOnClickHandler(id: "#negative-bars" | "#bars") {
	// ...
}
```

Reply:
Great suggestion.  I think there might be multiple event handlers I can DRY up. Jess mentioned we may also want to change the hover color of the label text on the bar and negative-bar mouseover so I'll incorporate that feedback with yours.